spark-bigquery-connector-common/third_party/apache-spark/src/main/java/com/google/cloud/spark/bigquery/AvroSchemaConverter.java

```diff
@@ -187,15 +194,33 @@ static Converter createConverterFor(DataType sparkType, Schema avroType) {
     if (sparkType instanceof DecimalType && avroType.getType() == Schema.Type.BYTES) {
       DecimalType decimalType = (DecimalType) sparkType;
       return (getter, ordinal) -> {
-        Decimal decimal = getter.getDecimal(ordinal, decimalType.precision(), decimalType.scale());
+        BigDecimal bigDecimal = null;
+        if (getter instanceof UnsafeRow || getter instanceof UnsafeArrayData) {
```

Comment:
Similar to the above question, were these changes required due to the direct write support?